lab-michaelstuart/route/user-router.js

```diff
@@ -0,0 +1,46 @@
+const User = require('../model/user');
+const router = require('../lib/router.js');
+
+router.post('/api/user', (req, res) => {
+  const { name, password, email } = req.body;
+  if (!name || !password || !email) return res.sendStatus(400);
+  new User(name, password, email)
+    .save()
+    .then(user => res.sendJSON(201, user))
+    .catch(() => res.sendStatus(500));
+});
+
+router.get('/api/user', (req, res) => {
+  const id = req.url.query.id;
+  if(!id) return res.sendStatus(400);
+  User.findById(id)
+    .then(user => res.sendJSON(200, user))
+    .catch(() => res.sendStatus(404));
+});
+
+router.put('/api/user', (req, res) => {
+  let { name, password, email } = req.body;
```

Comment:
Killer use of destructuring here. I haven't seen anyone else using this feature so nice work implementing it.